vault/templates/secret.yaml

```diff
@@ -1,24 +1,29 @@
 {{- if and (not .Values.tls.secretName) (not .Values.certManager.certificate.enabled) }}
-{{- $cn := include "vault.fullname" . -}}
+{{- $cn := include "vault.fullname" $ -}}
 {{- $externalDNS := print $cn "." .Release.Namespace -}}
 {{- $ca := genCA "vault-ca" 365 -}}
 {{- $cert := genSignedCert $cn (list "127.0.0.1") (list $cn $externalDNS) 365 $ca -}}
+{{- $caNamespaces := list .Release.Namespace -}}
+{{- if .Values.tls.caNamespaces }}{{ $caNamespaces = (concat $caNamespaces .Values.tls.caNamespaces ) | uniq }}{{ end -}}
+{{- range $i, $namespace := $caNamespaces }}
+{{- if $i }}{{ printf "\n---\n" }}{{ end -}}
```

Comment:
Conditional check is not needed, this should work by using `---` directly

Reply:
Done.